roofit/roofitcore/inc/RooFit/Detail/CodeSquashContext.h

```diff
@@ -101,17 +101,25 @@ class CodeSquashContext {
       const std::vector<TNamed const *> _vars;
    };
 
-   std::unique_ptr<LoopScope> beginLoop(RooArgSet const &loopVars);
+   std::unique_ptr<LoopScope> beginLoop(RooAbsArg const *in);
 
    std::string getTmpVarName();
 
+   std::string saveAsTemp(RooAbsArg const *in, std::string const &valueToSave, std::string name = "");
+
+   std::string saveListAsArray(RooListProxy const &in, std::string name = "");
+
 private:
    void endLoop(LoopScope const &scope);
 
-   void addResult(TNamed const *key, std::string const &value, bool isReducerNode);
+   void addResult(TNamed const *key, std::string const &value);
 
    std::string buildArg(double x) { return RooNumber::toString(x); }
 
+   std::string buildArg(std::string const &x) { return x; }
+
+   std::string buildArg(RooListProxy const &x) { return saveListAsArray(x); }
+
    template <class T>
    std::string buildArg(T const &arg)
    {
```

Comment:
I would replace the `buildArg` family with these overloads:
```C++
   std::string buildArg(double x) { return RooNumber::toString(x); }
   std::string buildArg(int x) { return std::to_string(x); }
   std::string buildArg(unsigned int x) { return std::to_string(x); }
   std::string buildArg(std::string const &x) { return x; }
   std::string buildArg(RooAbsCollection const &x) { return saveListAsArray(x); }
   std::string buildArg(RooAbsArg const &arg) { return getResult(arg); }
   template <class T>
   std::string buildArg(RooTemplateProxy<T> const &arg)
   {
      return getResult(arg);
   }
```
Like this, you don't have to cast `int` to `double` in the translate function, and don't get ambiguity errors from the `T const&` catch-all signature.

Also, please don't use `RooListProxy` but the collection base class `RooAbsCollection` (save in `saveListAsArray`).